### PR TITLE
[WIP] TVM compatible and Directpose

### PR DESCRIPTION
**Pull request by zhreshold**
WIP, adding TVM compatible pytorch implementation of Directpose

Changes made to existing models:

- changed the way of getting default configs, previously all default configs are retrieved with `get_cfg_defaults`, now it becomes `get_cfg_defaults(name="action_recognition")`, model or task-specific configs are retrieved via name. The default is "action_recognition" so only new models and their configs are affected.

Others:

- add tvm conversion and compile check in unittest(not enabled for all pre-trained models by default)
- added various utils and dataset functions for loading detection/keypoints datasets, mainly tested against coco format

### CODE REVIEW

**Comment by zhreshold** [comment]
PR tested and passed, please take a look @bryanyzhu

File: gluoncv/torch/data/registry/catalog.py [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+        assert len(name)
+        if name in MetadataCatalog._NAME_TO_META:
+            ret = MetadataCatalog._NAME_TO_META[name]
+            # TODO this is for the BC breaking change in D15247032.
+            # Remove this in the future.
+            if hasattr(ret, "dataset_name"):
+                logger = logging.getLogger()
+                logger.warning(
+                    """
+The 'dataset_name' key in metadata is no longer used for
```
Comment (bryanyzhu) [inline gluoncv/torch/data/registry/catalog.py:212]: The indentation looks weird, maybe it's better to put it within if statements.

File: gluoncv/torch/data/registry/metadata.py [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+    stuff_dataset_id_to_contiguous_id[0] = 0
+
+    # 54 names for COCO stuff categories (including "things")
+    stuff_classes = ["things"] + [
+        k["name"].replace("-other", "").replace("-merged", "")
+        for k in COCO_CATEGORIES
+        if k["isthing"] == 0
+    ]
+
+    # NOTE: I randomly picked a color for things
```
Comment (bryanyzhu) [inline gluoncv/torch/data/registry/metadata.py:225]: Is this consistent with other libraries or doesn't matter?

File: gluoncv/torch/data/structures/boxes.py [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+        Returns:
+            The box centers in a Nx2 array of (x, y).
+        """
+        return (self.tensor[:, :2] + self.tensor[:, 2:]) / 2
+
+    def scale(self, scale_x: float, scale_y: float) -> None:
+        """
+        Scale the box with horizontal and vertical scaling factors
+        """
+        # self.tensor[:, 0::2] *= scale_x
```
Comment (bryanyzhu) [inline gluoncv/torch/data/structures/boxes.py:277]: Should we remove the commented code?

File: gluoncv/torch/nn/nms.py [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+    boxes = boxlist.pred_boxes.tensor
+    scores = boxlist.scores
+    labels = boxlist.pred_classes
+    keep = batched_nms(boxes, scores, labels, nms_thresh)
+    # keep = nms(boxes, scores, nms_thresh)
+    if max_proposals > 0:
+        keep = keep[: max_proposals]
+    if fixed_size:
+        boxlist._keep = keep
+        # scores = -scores
```
Comment (bryanyzhu) [inline gluoncv/torch/nn/nms.py:54]: Multiple commented code blocks in this file. No need to do anything if you feel necessary.

**Review by bryanyzhu — APPROVED** [review]
Awesome PR! I left some minor comments. And there is a .gitignore file in `scripts/pose/directpose/.gitignore`, I don't know whether we should keep it. Thank you.

File: gluoncv/torch/data/registry/catalog.py (hunk 1 above)
Comment (zhreshold) [reply gluoncv/torch/data/registry/catalog.py]: fixed

File: gluoncv/torch/data/registry/metadata.py (hunk 2 above)
Comment (zhreshold) [reply gluoncv/torch/data/registry/metadata.py]: It's consistent

File: gluoncv/torch/data/structures/boxes.py (hunk 3 above)
Comment (zhreshold) [reply gluoncv/torch/data/structures/boxes.py:278]: I'd rather want to keep it as these are more readable, however to address tvm compatibility I need to use a concat op. added comments for the commented out code

File: gluoncv/torch/nn/nms.py (hunk 4 above)
Comment (zhreshold) [reply gluoncv/torch/nn/nms.py:55]: Added comments and keep these alternative implementation in case we need that later

**Pull request merged**